app/models/manageiq/providers/amazon/cloud_manager/refresh_worker.rb

```diff
@@ -14,7 +14,10 @@ def queue_name_for_ems(ems)
     private
 
     def combined_managers(ems)
-      [ems].concat(ems.child_managers)
+      child_managers = ems.child_managers.reject do |child_ems|
+        child_ems.kind_of?(ManageIQ::Providers::Amazon::StorageManager::S3)
```

Comment:
I'm okay with this for now, but wonder if we could have a more general way of solving this by adding a method on `ext_management_system` like `#combine_with_parent_manager?`

Reply:
I guess it requires more DRYing, since that method presents (exactly the same code) in other providers.
- https://github.com/ManageIQ/manageiq-providers-azure/blob/master/app/models/manageiq/providers/azure/cloud_manager/refresh_worker.rb#L16
- https://github.com/ManageIQ/manageiq-providers-openstack/blob/master/app/models/manageiq/providers/openstack/cloud_manager/refresh_worker.rb#L16

And I don't think it should be moved in an `ExtManagementSystem` because its only used there.

It can be probably moved here:
- https://github.com/ManageIQ/manageiq/blob/master/app/models/manageiq/providers/base_manager/refresh_worker.rb

And be overriden in subclasses as needed.

Reply:
Yeah makes sense to put this in the BaseManager::RefreshWorker